### PR TITLE
Importer refactoring

### PR DESCRIPTION
**Pull request by kprzerwa**

### CODE REVIEW

File: cds_ils/importer/importer.py [hunk 1]
```diff
@@ -154,82 +117,118 @@ def index_all_records(self):
...
+
         self._validate_provider()
-        self.action = "none"
-        # finds the exact match, update records
-        self.document = self._match_document()
 
-        if self.document:
-            self.output_pid = self.document["pid"]
-            self.delete_records(self.document)
+        exact_match, partial_matches = self._match_document()
```
Comment (ntarocco) [inline cds_ils/importer/importer.py:204]: In a future refactoring, this is also risky, same assumption that something happened before and maybe this `_match_document` will also set/change something in `self`.

File: cds_ils/importer/importer.py [hunk 2]
```diff
@@ -346,7 +363,7 @@ def preview_delete_document(self, document):
         related_refs = set()
         for _, related_objects in document.relations.items():
             for obj in related_objects:
-                if not obj["record_metadata"].get("mode_of_issuance")\
+                if not obj["record_metadata"].get("mode_of_issuance") \
```
Comment (ntarocco) [inline cds_ils/importer/importer.py:366]: is it ok to use `.get` here? can it be unset?

**Review by ntarocco — COMMENTED** [review]
It looks much better now ;)

**Review by ntarocco — APPROVED** [review]

File: cds_ils/importer/importer.py (hunk 2 above)
Comment (kprzerwa) [reply cds_ils/importer/importer.py:366]: there are other related record types which don't have the mode of issuance

**Pull request merged**